src/Utils/BlockTemplateUtils.php

```diff
@@ -134,25 +134,59 @@ public static function gutenberg_build_template_result_from_post( $post ) {
 	 */
 	public static function gutenberg_build_template_result_from_file( $template_file, $template_type ) {
 		$template_file = (object) $template_file;
+
+		// If the theme has an archive-products.html template but does not have product taxonomy templates
+		// then we will load in the archive-product.html template from the theme to use for product taxonomies on the frontend.
+		$template_is_from_theme = 'theme' === $template_file->source ? true : false;
+		$theme_name             = wp_get_theme()->get( 'Name' );
+
 		// phpcs:ignore WordPress.WP.AlternativeFunctions.file_get_contents_file_get_contents
 		$template_content         = file_get_contents( $template_file->path );
 		$template                 = new \WP_Block_Template();
-		$template->id             = 'woocommerce//' . $template_file->slug;
-		$template->theme          = 'woocommerce/woocommerce';
+		$template->id             = $template_is_from_theme ? strtolower( $theme_name ) . '//' . $template_file->slug : 'woocommerce//' . $template_file->slug;
+		$template->theme          = $template_is_from_theme ? $theme_name : 'woocommerce/woocommerce';
 		$template->content        = self::gutenberg_inject_theme_attribute_in_content( $template_content );
-		$template->source         = 'plugin';
+		$template->source         = $template_file->source || 'plugin';
```

Comment:
According to the [inline docs](https://github.com/WordPress/gutenberg/blob/8eb1f341285362f6a131fcba81a8d054b6b5b5c5/lib/compat/wordpress-5.9/class-gutenberg-block-template.php#L63-L68), `source` can only have two values: `theme` or `custom`. I guess for plugins we should simply use `theme`.

Reply:
In preparation for templates to be included for plugins we agreed we could use the `plugin` value for `source`. Please see this PR (https://github.com/woocommerce/woocommerce-gutenberg-products-block/pull/5215) and it's related issue where this was discussed and agreed (https://github.com/WordPress/gutenberg/issues/36597#issuecomment-976232909)

Reply:
Oh, I wasn't aware of that discussion, thanks for linking to it. :+1: